### PR TITLE
Add etcd package.

### PR DESCRIPTION
**Pull request by dlorenc**
Signed-off-by: Dan Lorenc <dlorenc@chainguard.dev>

### CODE REVIEW

File: etcd.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    packages:
+      - ca-certificates-bundle
+      - busybox
+      - git
+      - bash
+      - go
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://github.com/coreos/etcd/archive/v${{package.version}}.tar.gz
```
Comment (tuananh) [inline etcd.yaml:26]: nitpick: shouldn't it be https://github.com/etcd-io/etcd? coreos/etcd redirect to the new one anyway

File: etcd.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      expected-sha256: 0ef11c0cdc3cc693756a636d97d280ea63e222a7be38159fad0c8032ced0da42
+  - runs: |
+      bash -x ./build.sh
+      mkdir -p "${{targets.destdir}}"/var/lib/$pkgname
```
Comment (rawlingsj) [inline etcd.yaml:30]: `$pkgname` can change to the melange templated value `${{package.name}}` or just hard code `etcd`

File: etcd.yaml (hunk 1 above)
Comment (dlorenc) [reply etcd.yaml]: Fixed!

File: etcd.yaml (hunk 2 above)
Comment (dlorenc) [reply etcd.yaml]: Fixed!

Comment (imjasonh) [reply etcd.yaml]: I've definitely made this mistake before. We should add a check for this in linting.

**Review by imjasonh — APPROVED** [review]

**Review by jedsalazar — APPROVED** [review]

**Review by puerco — APPROVED** [review]

**Pull request merged**